server.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+try {
+  // $DBH = new PDO("mysql:host=$host;dbname=$dbname", $user, $pass); //MYSQL database
+  $conn = new PDO("sqlite:/etc/pihole/pihole.db");  // SQLite Database
+  $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+
+  $hostname = trim(file_get_contents("/etc/hostname"), "\x00..\x1F");
+
+  $where ="(source || name !='127.0.0.1$hostname')";
+  $order_by="id";
+  $rows=25;
+  $current=1;
+  $limit_l=($current * $rows) - ($rows);
+  $limit_h=$limit_l + $rows  ;
+
+
+//Handles Sort querystring sent from Bootgrid
+  if (isset($_REQUEST['sort']) && is_array($_REQUEST['sort']) )
+  {
+    $order_by="";
+    foreach($_REQUEST['sort'] as $key=> $value)
+      $order_by.=" $key $value";
+  }
+
+//Handles search  querystring sent from Bootgrid
+  if (isset($_REQUEST['searchPhrase']) )
+  {
+    $search=trim($_REQUEST['searchPhrase']);
+    $where.= " AND ( ts LIKE '%".$search."%' OR  query_type LIKE '%".$search."%' OR  name LIKE '%".$search."%' OR  source LIKE '%".$search."%' OR domain LIKE '%".$search."%'  ) ";
```

Comment:
Going to need to sanitise this a bit to stop any chance of SQL injection. I think the only value that can be user-input is $search